src/composables/useTokenGenerator/useTokenGenerator.ts

```diff
@@ -66,23 +74,59 @@ export default function useTokenGenerator() {
 
     /**
      * Generate a valid token using the rules defined in the rules object
-     * @returns a valid token
+     * @param {number} length - The length of the token (default: 32)
+     * @param {string} alphabet - The set of characters used to generate the token (default: characterSet)
+     * @param {boolean} isLowerCase - Indicates whether to include lowercase letters in the token (default: true)
+     * @param {boolean} isUpperCase - Indicates whether to include uppercase letters in the token (default: true)
+     * @param {boolean} isSpecial - Indicates whether to include special characters in the token (default: true)
+     * @param {boolean} isNumber - Indicates whether to include numbers in the token (default: true)
+     * @param {boolean} isMin - Indicates whether the token must satisfy a minimum requirement of having at least one character from each selected category (default: true)
+     * @returns {string} - A valid token
      */
-    const generateValidToken = (): string => {
-        const token = generateToken();
-        if (
-            defaultRules.lowercase(token) === true &&
-            defaultRules.uppercase(token) === true &&
-            defaultRules.special(token) === true &&
-            defaultRules.number(token) === true &&
-            defaultRules.min(token) === true
-        ) {
+    const generateValidToken = (
```

Comment:
I guess here lies the issue regarding upper comment. length is not being checked for